### PR TITLE
fancier dev tools

### PR DESCRIPTION
**Pull request by nicolaskruchten**
* Ace editor for full control over input JSON, including copy-paste in and out.
* lazy-loading inspector for `_fullData` and `_fullLayout` to be able to drill into circular structures

### CODE REVIEW

File: dev/App.js [hunk 1]
```diff
@@ -3,9 +3,13 @@ import {hot} from 'react-hot-loader';
 import plotly from 'plotly.js/dist/plotly';
 import '../src/styles/main.scss';
 import 'react-select/dist/react-select.css';
-import ReactJson from 'react-json-view';
+import brace from 'brace'; // eslint-disable-line no-unused-vars
+import AceEditor from 'react-ace';
```
Comment (VeraZab) [inline dev/App.js]: can we do `import 'brace'` ?

we're loading those 2 resources down below:
import 'brace/mode/json';
import 'brace/theme/textmate';

that's not all we need?

File: dev/App.js [hunk 2]
```diff
@@ -59,15 +65,32 @@ class App extends Component {
...
       });
   }
 
+  updateState(data, layout, frames, currentMockIndex) {
+    this.setState({
+      data,
+      layout,
+      frames,
+      currentMockIndex,
+      full: 'hit refresh',
```
Comment (VeraZab) [inline dev/App.js]: there's `mockIndex` in the state above, are `mockIndex` and `currentMockIndex` meant to be the same?

File: package.json [hunk 3]
```diff
@@ -104,18 +103,25 @@
...
     "test:js": "jest --setupTestFrameworkScriptFile=raf/polyfill",
-    "test:lint": "eslint \"src/**/*.js\" && echo -e '\\033[0;32m'PASS'\\033[0m'",
-    "test:pretty": "prettier -l \"src/**/*.js\" && echo -e '\\033[0;32m'PASS'\\033[0m'",
-    "watch": "babel src --watch --out-dir lib --source-maps | node-sass -w src/styles/main.scss lib/react-chart-editor.css",
+    "test:lint":
+      "eslint \"src/**/*.js\" && echo -e '\\033[0;32m'PASS'\\033[0m'",
+    "test:pretty":
+      "prettier -l \"src/**/*.js\" && echo -e '\\033[0;32m'PASS'\\033[0m'",
+    "watch":
+      "babel src --watch --out-dir lib --source-maps | node-sass -w src/styles/main.scss lib/react-chart-editor.css",
```
Comment (VeraZab) [inline package.json]: I see your prettier for json is acting up now too :)

**Comment by VeraZab** [comment]
well, looks good 💃 except maybe that little state typo for currentMockIndex?

File: dev/App.js (hunk 1 above)
Comment (nicolaskruchten) [reply dev/App.js]: I'm not sure. I followed the readme from `react-ace`

File: dev/App.js (hunk 2 above)
Comment (nicolaskruchten) [reply dev/App.js]: The state key is indeed `currentMockIndex`, so this is correct, but in a function above we get a new index called `mockIndex`, which we then use to overwrite.

File: package.json (hunk 3 above)
Comment (nicolaskruchten) [reply package.json]: gah. will fix. I reset my project to try to shake the CSS bug and lost my local settings

**Pull request merged**